### PR TITLE
Update macOS packaging for py3

### PR DESCRIPTION
**Pull request by palango**
Don't use detox as it pulls enum34.
Simplify some of the macOS setup with homebrew.

This is a part of #1205

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -181,7 +180,7 @@ def read_version_from_git():
     },
     install_requires=install_requires,
     tests_require=test_requirements,
-    python_requires='>=3.5',
+    python_requires='>=3.6',
```
Comment (hackaugusto) [inline setup.py]: What are the breaking changes that make it impossible to run this project with the version 3.5 of the interpreter?

File: requirements-dev.txt [hunk 2]
```diff
@@ -8,7 +8,7 @@ pytest-cov==2.5.1
 flaky==3.4.0
 hypothesis==3.44.14
 grequests==0.3.0
-detox==0.11
+tox==2.9.1
```
Comment (hackaugusto) [inline requirements-dev.txt]: we could also remove this dependency, since it's not really used by any of our scripts

Comment (palango) [reply requirements-dev.txt]: Ok

File: setup.py (hunk 1 above)
Comment (palango) [reply setup.py]: It's the `encoding` argument in the `Popen`call:
> New in version 3.6: encoding and errors were added.

**Pull request merged**